packages/database/zod/insights.ts

```diff
@@ -1,30 +1,33 @@
+import { type Insight, InsightCategory } from "@prisma/client";
 import { z } from "zod";
-import { ZId } from "./common";
+import { ZId } from "../../types/common";
 
-export const ZInsightCategory = z.enum(["featureRequest", "complaint", "praise", "other"]);
+export const ZInsightCategory = z.nativeEnum(InsightCategory);
 
-export type TInsightCategory = z.infer<typeof ZInsightCategory>;
-
-export const ZInsight = z.object({
+const InsightModel = z.object({
   id: ZId,
   createdAt: z.date(),
   updatedAt: z.date(),
   environmentId: ZId,
   title: z.string(),
   description: z.string(),
   category: ZInsightCategory,
+}) satisfies z.ZodType<Insight>;
+
+export const ZInsight = InsightModel.extend({
   _count: z.object({
     documentInsights: z.number(),
   }),
 });
 
 export type TInsight = z.infer<typeof ZInsight>;
 
-export const ZInsightCreateInput = z.object({
-  environmentId: ZId,
-  title: z.string(),
-  description: z.string(),
-  category: ZInsightCategory,
+export const ZInsightCreateInput = InsightModel.pick({
```

Comment:
Not sure if this should be part of this file. Since it's more an application decision what data to require as input for this use case it should be moved to the application where this is needed.